src/layers/src/layer-utils.ts

```diff
@@ -34,3 +37,83 @@ export function assignPointPairToLayerColumn(pair: FieldPair, hasAlt: boolean) {
     altitude: alt ? {...defaultAltColumn, ...alt} : defaultAltColumn
   };
 }
+
+export function getGeojsonLayerMetaFromArrow({
+  dataContainer,
+  getColumn,
+  getField
+}: {
+  dataContainer: DataContainerInterface;
+  getColumn: (dataContainer?: DataContainerInterface) => any;
+  getField: (dataContainer?: DataContainerInterface) => any;
+}) {
+  const geoColumn: arrow.Vector = getColumn(dataContainer);
+  const arrowField: Field = getField(dataContainer);
+
+  const encoding = arrowField?.metadata?.get('ARROW:extension:name');
+  // create binary data from arrow data for GeoJsonLayer
+  const {binaryGeometries, featureTypes, bounds} = getBinaryGeometriesFromArrow(
+    geoColumn,
+    encoding
+  );
+
+  // since there is no feature.properties.radius, we set fixedRadius to false
+  const fixedRadius = false;
+
+  return {
+    dataToFeature: binaryGeometries,
+    featureTypes,
+    bounds,
+    fixedRadius
+  };
+}
+
+export function isLayerHoveredFromArrow(objectInfo, layerId) {
+  // there could be multiple deck.gl layers created from multiple chunks in arrow table
+  // the objectInfo.layer id should be `${this.id}-${i}`
+  if (objectInfo?.picked) {
+    const deckLayerId = objectInfo?.layer?.props?.id;
+    return deckLayerId.startsWith(layerId);
+  }
+  return false;
+}
+
+export function getHoveredObjectFromArrow(
```

Comment:
Could part of this (getting an object from a table) somehow go into data container interface?

Reply:
Yes, I believe this can be consolidated when working on supporting arrow in other layers.